cmake/FindPoly.cmake

```diff
@@ -143,6 +143,14 @@ if(NOT Poly_FOUND_SYSTEM)
       "${DEPS_BASE}/lib/libpicpolyxx${CMAKE_STATIC_LIBRARY_SUFFIX}")
   endif()
 
+  if(NOT(WASM STREQUAL "OFF"))
+    set(POLY_PATCH_CMD ${POLY_PATCH_CMD}
+      COMMAND
+        patch -i ${CMAKE_SOURCE_DIR}/cmake/deps-utils/Poly-wasm.patch 
+                 ${DEPS_BASE}/src/Poly-EP/src/upolynomial/factorization.c
+    )
+  endif()
+
```

Comment:
Try to substitute this patch by inserting a global flag that avoids the abortion of the compilation process when detecting `declared but unused variables`.